src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java

```diff
@@ -505,4 +510,91 @@ void shouldReturnTransactionsCorrectlyForDateRangesAndNumberOfTransactions() {
             ));
         }
     }
+
+    @Nested
+    @DisplayName("RedactPIIFromTransaction")
+    class TestRedactPIIFromTransaction {
+
+        @Test
+        void shouldRedactAllPIIFromTransactionCorrectly() throws JsonProcessingException {
+
+            JsonObject transactionDetails1 = new JsonObject();
+            transactionDetails1.addProperty("reference", "ref-1");
+            transactionDetails1.addProperty("cardholder_name", "Joe B");
+            transactionDetails1.addProperty("email", "test@email.com");
+            transactionDetails1.addProperty("address_line1", "line 1");
+            transactionDetails1.addProperty("address_line2", "line 2");
+
+            TransactionEntity transactionToRedact = aTransactionFixture()
+                    .withReference("ref-1")
+                    .withEmail("test@email.com")
+                    .withCardholderName("Joe B")
+                    .withTransactionDetails(transactionDetails1.toString())
+                    .insert(rule.getJdbi())
+                    .toEntity();
+
+            JsonObject transactionDetails2 = new JsonObject();
+            transactionDetails2.addProperty("reference", "ref-2");
+            transactionDetails2.addProperty("cardholder_name", "Jane D");
+            transactionDetails2.addProperty("email", "test2@email.com");
+            TransactionEntity transactionThatShouldNotBeRedacted = aTransactionFixture()
```

Comment:
What makes this a transaction that should not be redacted?

Reply:
DAO method is to redact PII for a single transaction. This is another transaction that should not be selected for redaction.

Reply:
Ahhh ok.